File: dashlane-project/modules/base-ui/src/main/java/com/dashlane/ui/adapter/HeaderItem.java
package com.dashlane.ui.adapter;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

import com.dashlane.ui.R;
import com.dashlane.ui.adapter.util.DiffUtilComparator;
import com.skocken.efficientadapter.lib.viewholder.EfficientViewHolder;

import java.util.Objects;

import androidx.annotation.NonNull;
import androidx.core.view.ViewCompat;



public class HeaderItem
        implements DashlaneRecyclerAdapter.MultiColumnViewTypeProvider,
        DiffUtilComparator<HeaderItem> {

    private static final DashlaneRecyclerAdapter.ViewType<? extends HeaderItem> VIEW_TYPE =
            new DashlaneRecyclerAdapter.ViewType<>(R.layout.item_header, HeaderItemViewHolder.class);

    private final String mLabel;

    public HeaderItem(String label) {
        mLabel = label;
    }

    @Override
    public int getSpanSize(int spanCount) {
        return spanCount;
    }

    public String getLabel() {
        return mLabel;
    }

    @NonNull
    @Override
    public DashlaneRecyclerAdapter.ViewType getViewType() {
        return VIEW_TYPE;
    }

    @Override
    public boolean isItemTheSame(HeaderItem item) {
        return Objects.equals(mLabel, item.mLabel);
    }

    @Override
    public boolean isContentTheSame(HeaderItem item) {
        return isItemTheSame(item);
    }

    public static class HeaderItemViewHolder extends EfficientViewHolder<HeaderItem> {

        public HeaderItemViewHolder(View itemView) {
            super(itemView);
        }

        @Override
        protected void updateView(@NonNull Context context, HeaderItem object) {
            TextView textView = (TextView) getView();
            textView.setText(object.mLabel);
            ViewCompat.setAccessibilityHeading(textView, true);
        }

        @Override
        public boolean isClickable() {
            return false;
        }
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/notificationcenter/view/HeaderItem.kt
package com.dashlane.notificationcenter.view

import android.content.Context
import android.view.View
import com.dashlane.R
import com.dashlane.ui.adapter.DashlaneRecyclerAdapter
import com.dashlane.ui.adapter.util.DiffUtilComparator
import com.skocken.efficientadapter.lib.viewholder.EfficientViewHolder

data class HeaderItem(
    val section: ActionItemSection,
    var count: Int?,
    val seeAllListener: (ActionItemSection) -> Unit
) : DashlaneRecyclerAdapter.MultiColumnViewTypeProvider, DiffUtilComparator<HeaderItem> {
    override fun getViewType(): DashlaneRecyclerAdapter.ViewType<*> {
        return VIEW_TYPE
    }

    override fun getSpanSize(spanCount: Int): Int = spanCount

    override fun isItemTheSame(item: HeaderItem) = section == item.section

    override fun isContentTheSame(item: HeaderItem) = this == item

    companion object {
        val VIEW_TYPE = DashlaneRecyclerAdapter.ViewType(
            R.layout.item_actionitem_header,
            ViewHolder::class.java
        )
    }

    class ViewHolder(view: View) : EfficientViewHolder<HeaderItem>(view) {
        override fun updateView(context: Context, item: HeaderItem?) {
            item ?: return
            setText(R.id.title, context.getString(item.section.titleRes))
            val seeAllText = if (item.count != null) {
                context.getString(R.string.section_header_see_all_button_with_count, item.count)
            } else {
                context.getString(R.string.section_header_see_all_button)
            }
            setText(
                R.id.see_all_button,
                seeAllText
            )
            findViewByIdEfficient<View>(R.id.see_all_button)?.setOnClickListener { item.seeAllListener(item.section) }
        }
    }
}